src/main/resources/sql/maxwell_schema.sql

```diff
@@ -1,34 +1,58 @@
-CREATE DATABASE IF NOT EXISTS `maxwell`;
+SET @c = CONCAT('CREATE DATABASE IF NOT EXISTS ', @maxwell_db)
 
-CREATE TABLE IF NOT EXISTS `maxwell`.`schemas` (
+PREPARE maxwell_stmt from @c
+
+EXECUTE maxwell_stmt
+
+DEALLOCATE PREPARE maxwell_stmt
+
+SET @c = CONCAT('CREATE TABLE IF NOT EXISTS `', @maxwell_db, '`.`schemas` (
```

Comment:
I have setCatalog working, but I ran into an odd problem when running the tests.

When I run the tests some of the them would fail with a no database selected SQL Exception. However when I rerun just the failed test, it would pass. It turned out that CachedStatements didn't maintain the database from setCatalog call. If I disable caching on the connection pool then the tests pass except for the case sensitivity test.

I am going to merge master into this branch before proceeding since my previously merged PR will conflict with this branch.